addons/mass_mailing/models/mailing.py

```diff
@@ -970,7 +970,27 @@ def _get_mass_mailing_context(self):
 
     def _get_recipients(self):
         mailing_domain = self._parse_mailing_domain()
-        res_ids = self.env[self.mailing_model_real].search(mailing_domain).ids
+        res_model = self.env[self.mailing_model_real]
+        if "email" in res_model._fields:
+            res_ids_grouped = res_model.read_group(
+                mailing_domain, ["__my_id:max(id)"], "email"
+            )
+            res_ids = [r["__my_id"] for r in res_ids_grouped]
+
+        elif "partner_id" in res_model._fields:
+            res_ids_grouped = res_model.read_group(
+                mailing_domain, ["__my_id:max(id)"], "partner_id", orderby="partner_id"
```

Comment:
Could you double-confirm if the list is not needed here?

```suggestion
                mailing_domain, ["__my_id:max(id)"], ["partner_id"], orderby="partner_id"
```

Reply:
![image](https://github.com/user-attachments/assets/f911b6a8-0ff7-44e8-ad85-f9b19f020006)

Looks like either is fine

Reply:
Yes, it looks as Odoo is transforming the values from str to list but before transforming OrderedSet 🤣 
 - https://github.com/Vauxoo/odoo/blob/5d4a24afa983a82357d76b5d6a09750b26ce3de7/odoo/models.py#L2234

Could you use directly a list in order to avoid recomputing this extra transformations, please?

Reply:
Forget it

It looks like the list is transformed but not for str WTF

Reply:
Line 2234 is probably a way to remove duplicates, in the unlikely scenario that one would group by the same field twice? So I leave it as is?

Reply:
I prefer using the list since that it is the expected value in the most of cases

But I don't have a good justification for that

Maybe when it is migrated to use typing we will have issues since that only one type of data will be compatible

Reply:
@moylop260 Done!